### PR TITLE
fixes  #239

### LINKED ISSUE #239: Dismiss desktop notification on right click
Desktop notifications (those which usually pop up in the bottom right corner) should be dismissable by right-clicking on them.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/35160626-dismiss-desktop-notification-on-right-click?utm_campaign=plugin&utm_content=tracker%2F14200913&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F14200913&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by axel1200**

### CODE REVIEW

**Comment by axel1200** [comment]
You mean implementing it in com.faforever.client.notification.TransientNotificationControllerTest?

**Comment by micheljung** [comment]
Yes

**Comment by axel1200** [comment]
Travis fail is not related to changes made

File: src/main/java/com/faforever/client/notification/TransientNotificationController.java [hunk 1]
```diff
@@ -107,6 +109,10 @@ public Region getRoot() {
   }
 
   public void onClicked(MouseEvent event) {
-    action.call(event);
+    if (event.getButton().equals(MouseButton.SECONDARY)) {
+      dismiss();
+    } else {
+      action.call(event);
+    }
```
Comment (micheljung) [inline src/main/java/com/faforever/client/notification/TransientNotificationController.java]: Can we have a unit test for this, which checks that the correct thing happens based on which mouse button has been clicked?

Comment (axel1200) [inline src/main/java/com/faforever/client/notification/TransientNotificationController.java]: I do allready have testOnRightClick() and I will add testOnLeftClick() ;)

Comment (axel1200) [inline src/main/java/com/faforever/client/notification/TransientNotificationController.java]: done

**Comment by axel1200** [comment]
And how do I do this? With rebase?

**Comment by micheljung** [comment]
IntelliJ: VCS -> Git -> Rebase -> Select "interactive" -> Squash all commits but the last one, reword the last one

basically :)
Alternatively; Do a soft reset to your first commit, then commit mit "amend" selected.

In both cases, force push

**Comment by axel1200** [comment]
Haha ok that's probably not what u wanted. That's what happend when I did soft reset. Unlucköy I did not manage rebase.

**Comment by micheljung** [comment]
Let's try to fix this together. Today?

**Pull request merged**